contrib/jackc/pgx.v5/metrics.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package pgx
+
+import (
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+const tracerPrefix = "datadog.tracer."
+
+const (
+	AcquireCount            = tracerPrefix + "pgx.pool.connections.acquire"
+	AcquireDuration         = tracerPrefix + "pgx.pool.connections.acquire_duration"
+	AcquiredConns           = tracerPrefix + "pgx.pool.connections.acquired_conns"
+	CanceledAcquireCount    = tracerPrefix + "pgx.pool.connections.canceled_acquire"
+	ConstructingConns       = tracerPrefix + "pgx.pool.connections.constructing_conns"
+	EmptyAcquireCount       = tracerPrefix + "pgx.pool.connections.empty_acquire"
+	IdleConns               = tracerPrefix + "pgx.pool.connections.idle_conns"
+	MaxConns                = tracerPrefix + "pgx.pool.connections.max_conns"
+	TotalConns              = tracerPrefix + "pgx.pool.connections.total_conns"
+	NewConnsCount           = tracerPrefix + "pgx.pool.connections.new_conns"
+	MaxLifetimeDestroyCount = tracerPrefix + "pgx.pool.connections.max_lifetime_destroy"
+	MaxIdleDestroyCount     = tracerPrefix + "pgx.pool.connections.max_idle_destroy"
+)
+
+var interval = 10 * time.Second
+
+// pollPoolStats calls (*pgxpool).Stats on the pool at a predetermined interval. It pushes the pool Stats off to the statsd client.
+func pollPoolStats(statsd internal.StatsdClient, pool *pgxpool.Pool) {
+	if pool == nil {
+		log.Debug("No traced pool connection found; cannot pull pool stats.")
+		return
+	}
```

Comment:
We can remove lines 38 - 41 as they essentially duplicate the `err != nil` check done by the caller, `NewPoolWithConfig` (Note: We also removed a similar check in the database/sql DB Stats feature in more recent versions , [PR](https://github.com/DataDog/dd-trace-go/pull/2682/files))